meshtastic/tcp_interface.py

```
@@ -347,12 +353,27 @@ def _write_bytes(self, b: bytes) -> None:
+            total_sent = 0
+            write_deadline = time.monotonic() + WRITE_PROGRESS_TIMEOUT_SECONDS
+            while total_sent < len(payload):
+                remaining = write_deadline - time.monotonic()
+                if remaining <= 0:
+                    raise TimeoutError(self.WRITE_TIMEOUT_ERROR)
+                _, writable, _ = select.select([], [sock], [], remaining)
+                if not writable:
+                    raise TimeoutError(self.WRITE_TIMEOUT_ERROR)
+                sent = sock.send(payload[total_sent:])
+                if sent <= 0:
+                    raise OSError("TCP write returned no bytes")
```

Ruff (0.15.2)

[warning] Abstract `raise` to an inner function

(TRY301)